src/xcommon.cpp

```diff
@@ -186,6 +197,69 @@ namespace xw
         m_comm.close(nl::json::object(), nl::json::object(), xeus::buffer_sequence());
     }
 
+    namespace
+    {
+        std::string tolower(std::string s)
+        {
+            auto safe_tolower = [](unsigned char c)
+            {
+                return std::tolower(c);
+            };
+            std::transform(s.begin(), s.end(), s.begin(), safe_tolower);
+            return s;
+        }
+
+        std::string ltrim(std::string s)
+        {
+            auto const safe_isnotspace = [](unsigned char ch)
+            {
+                return !std::isspace(ch);
+            };
+            s.erase(s.begin(), std::find_if(s.begin(), s.end(), safe_isnotspace));
+            return s;
+        }
+
+        std::string rtrim(std::string s)
+        {
+            auto const safe_isnotspace = [](unsigned char ch)
+            {
+                return !std::isspace(ch);
+            };
+            s.erase(std::find_if(s.rbegin(), s.rend(), safe_isnotspace).base(), s.end());
+            return s;
+        }
+
+        std::string trim(std::string s)
+        {
+            rtrim(s);
+            ltrim(s);
+            return s;
+        }
+
+        bool is_true_string(const char* str)
+        {
+            const std::string s = tolower(trim(str));
+            static const auto trues = {"true", "on", "yes", "1"};
```

Comment:
I'm curious, why aren't we mirroring the logic in [Jupyter Widgets](https://github.com/jupyter-widgets/ipywidgets/blob/7325e5952efb71bd69692b2d7ed815646c0ac521/python/ipywidgets/ipywidgets/widgets/widget.py#L36) and [Jupyter Core](https://github.com/jupyter/jupyter_core/blob/98ab1ef453956333a85bb6eee494ad0a9bab2c02/jupyter_core/paths.py#L47) of testing for the absence of negative values instead of the presence of positive values?

Reply:
Also in Jupyter Widgets and Jupyter Core, the search is case insensitive, so `N`, `False`, etc. counts as a negative value too.

Reply:
Indeed there is no reason. I did not have all that context at the time of writing, we can change that.